src/identity/components/MarketoAccountUpgradeOverlay.tsx

```diff
@@ -0,0 +1,227 @@
+
+export const MarketoAccountUpgradeOverlay: FC = () => {
+  const {onClose} = useContext(OverlayContext)
+  const dispatch = useDispatch()
+
+  const user = useSelector(selectUser)
+  const accountId = useSelector(selectCurrentAccountId)
+
+  const [scriptHasLoaded, setScriptHasLoaded] = useState(false)
+  const [formHasLoaded, setFormHasLoaded] = useState(false)
+
+  const handleCloseOverlay = () => {
+    // Deleting marketo object guards against inconsistent behavior if user closes then re-opens the overlay.
+    delete window.MktoForms2
+    onClose()
+  }
+
+  const handleError = useCallback(
+    (message: MarketoError, err: Error = new Error(message)) => {
+      if (message === MarketoError.FormSubmitError) {
+        dispatch(notify(marketoFormSubmitFailure(SalesFormLink)))
+      } else {
+        dispatch(notify(marketoLoadFailure(SalesFormLink)))
+      }
+
+      reportErrorThroughHoneyBadger(err, {
+        context: {
+          user,
+          accountId,
+        },
+        name: message,
+      })
+    },
+    [accountId, dispatch, user]
+  )
+
+  const handleFormSubmission = () => {
+    try {
+      window.MktoForms2.getForm(MARKETO_FORM_ID).submit()
+      dispatch(notify(marketoFormSubmitSuccess()))
+      onClose()
+    } catch (err) {
+      handleError(MarketoError.FormSubmitError, err)
+    }
+  }
+
+  const loadMarketoForm = useCallback(() => {
+    try {
+      if (!formHasLoaded) {
+        // API: https://developers.marketo.com/javascript-api/forms/api-reference/
+        window.MktoForms2?.loadForm(
+          MARKETO_SERVER_INSTANCE,
+          MARKETO_SUBSCRIPTION_ID,
+          MARKETO_FORM_ID,
+          () => {
+            // Marketo fields should remain hidden from user (CSS display: false)
+
+            const marketoForm = window.MktoForms2
+            if (marketoForm) {
+              // Autofill the form with the current user and account id.
+              marketoForm.getForm(MARKETO_FORM_ID).setValues({
+                Quartz_User_ID__c: parseInt(user.id),
+                Quartz_Account_ID__c: accountId,
+              })
+
+              // Return false after form submission to prevent page reload.
+              marketoForm.whenReady((form: any) => {
+                form.onSuccess(() => false)
+              })
+
+              document.querySelectorAll('input').forEach(input => {
+                input.readOnly = true
+                input.disabled = true
+              })
+
+              setFormHasLoaded(true)
+            }
+          }
+        )
+      }
+    } catch (err) {
+      handleError(MarketoError.FormLoadingError, err)
+    }
+  }, [accountId, formHasLoaded, handleError, user])
+
+  useEffect(() => {
+    try {
+      if (!scriptHasLoaded) {
+        const marketoScript = document.createElement('script')
+        marketoScript.id = 'marketoScript'
+        marketoScript.src = `${MARKETO_SERVER_INSTANCE}/js/forms2/js/forms2.min.js`
+        marketoScript.onload = loadMarketoForm
```

Comment:
do we need to add

```ts
marketoScript.async = true
```
here?

Reply:
Yes, I'll add that - you're right that we want the script load to be non-blocking.